types/request.d.ts

```diff
@@ -22,21 +22,27 @@ export interface FastifyRequest<
   params: RouteGeneric['Params'];
   raw: RawRequest;
   query: RouteGeneric['Querystring'];
-  headers: RawRequest['headers'] & RouteGeneric['Headers']; // this enables the developer to extend the existing http(s|2) headers list
   log: FastifyLoggerInstance;
   body: RouteGeneric['Body'];
-  ip: string;
-  ips?: string[];
-  hostname: string;
-  protocol: 'http' | 'https';
-  url: string;
-  method: string;
-  routerPath: string;
-  routerMethod: string;
-  is404: boolean;
+
   /** in order for this to be used the user should ensure they have set the attachValidation option. */
   validationError?: Error & { validation: any; validationContext: string };
 
+  /**
+   * @deprecated Use `raw` property
+   */
+  readonly req: RawRequest;
+  readonly headers: RawRequest['headers'] & RouteGeneric['Headers']; // this enables the developer to extend the existing http(s|2) headers list
+  readonly ip: string;
```

Comment:
So is this still readonly?

Reply:
It is https://github.com/fastify/fastify/blob/master/lib/request.js#L135